packages/patternfly-3/react-console/src/DesktopViewer/DesktopViewer.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Grid, Row, Col } from 'patternfly-react';
+
+import ManualConnection from './ManualConnection';
+import ConnectWithRemoteViewer from './ConnectWithRemoteViewer';
+import consoleDetailPropType from './consoleDetailPropType';
+
+const DesktopViewer = ({ children, spice, vnc, onGenerate, onDownload, ...props }) => (
+  <Grid fluid className="desktop-viewer">
+    <Row>
+      <Col md={6}>
+        <ConnectWithRemoteViewer spice={spice} vnc={vnc} onGenerate={onGenerate} onDownload={onDownload}>
```

Comment:
can't we just detect the type here instead of checking everywhere on inner components? e.g. lets just provide a ```type=``` and render a child component based on the actual type passed?

Reply:
We need to pass both types down for `ManualConnection`.

For the `ConnectWithRemoteViewer`, the decision is just in the onClick button's handler.
The "spice over vnc" preference is hardcoded there, but I consider it as a part of the pattern.

This straightforward listing might become long if adding new types but recently just RDP is under consideration.

Since we probably need to pass the type back in the `onGenerate()` call-back, I don't see simplification in doing that - still same (2) props.